work16/src/main.js

```diff
@@ -0,0 +1,292 @@
+
+  return result;
+}
+
+const clickEventChangeTabAttribute = () => {
+  const tabTopics = document.querySelectorAll(".tabTopics");
+  for (let i = 0; i < tabTopics.length; i++) {
+    tabTopics[i].addEventListener("click", (event) => {
+      const selectedTab = document.querySelector('[aria-selected="true"]');
+      selectedTab.ariaSelected = false;
+      event.currentTarget.ariaSelected = true;
+    });
+  }
+}
+
+const removeCommentIcon = () => {
+  const commentIconWrapElements = document.querySelectorAll(".commentIcon_wrap");
+  for (let i = 0; i < commentIconWrapElements.length; i++) {
+    commentIconWrapElements[i].remove();
+  }
+}
+
+const removeNewIcon = () => {
+  const newIconElements = document.querySelectorAll(".newIcon");
+  for (let i = 0; i < newIconElements.length; i++) {
+    newIconElements[i].remove();
+  }
+}
+
+const changeCategoryTitle = (newsArticleData, categoryNewsArticleDataIndex) => {
+  const currentNewsContentAncorElements = document.querySelectorAll(`section[roll="tabpanel"] a`);
+  currentNewsContentAncorElements[categoryNewsArticleDataIndex].textContent = newsArticleData.title;
+}
+
+const changeCategoryImg = (newsCategoryImg) => {
+  const currentNewsContentImgElem = document.querySelector(".tabTopicImg > img");
+  currentNewsContentImgElem.src = newsCategoryImg;
+}
+
+const changeCategoryConetent = (newsCategoryIndex, clickedTabElement) => {
+  const tabTopicIdName = clickedTabElement.id;
+  const currentNewsSectionElem = document.querySelector(`section[roll="tabpanel"]`);
+
+  currentNewsSectionElem.id = `${"tabpanelTopics" + (newsCategoryIndex + 1)}`;
+  currentNewsSectionElem.setAttribute("aria-labelledby", tabTopicIdName);
+}
+
+const renderArticle = (newsContentsData) => {
+  const ul = document.createElement("ul");
+  const fragment = document.createDocumentFragment();
+
+  newsContentsData.forEach(newsArticleData => {
+    const li = document.createElement("li");
+    const h1 = document.createElement("h1");
+    const a = document.createElement("a");
+
+    const title = newsArticleData.title;
+    const commentArray = newsArticleData.comments;
+
+    a.href = "#";
+    a.textContent = title;
+
+    h1.appendChild(a)
+    li.appendChild(h1);
+
+    if (commentArray.length !== 0) {
+      li.append(createCommentIcon(commentArray));
+    }
+
+    if (isLatestArticles(newsArticleData) === true) {
+      h1.insertAdjacentElement("afterend", createNewIcon());
+    }
+
+    fragment.appendChild(li);
+  })
+  ul.appendChild(fragment);
+
+  return ul;
+}
+
+const createCommentIcon = (commentArray) => {
+  const div = document.createElement("div");
+
+  div.classList = "commentIcon_wrap";
+  div.insertAdjacentHTML("beforeend",
+    `<span class="fa-layers fa-fw">
+            <i class="fa fa-light fa-comment faa-vertical animated"></i>
+            <span class= "fa-layers-counter" > ${commentArray.length}</span>
+        </span>`
+  );
+
+  return div;
+}
```

Comment:
IMO
I think that the comment icon also needs to be created with createElement.

> htmlはulだけ作ってあとはcreateElementで作る

Reply:
I wasn't sure if adding a class with createElement would work to use FontAwesome's animation feature, so I used insertAdjacentHtml.
I used createElement and added the class and it worked.
It was a great learning experience. Thank you very much.
Thank you Mai-san.

commit: [96a13ba](https://github.com/yasu-pro/js-handson/pull/18/commits/96a13ba53f30fd3cc0644197a4f571456df31ae6)